packages/lesswrong/lib/helpers.ts

```diff
@@ -52,9 +52,12 @@ Utils.getUnusedSlugByCollectionName = async function (collectionName: Collection
   return await Utils.getUnusedSlug(collection, slug, useOldSlugs, documentId)
 };
 
-Utils.slugIsUsed = async (collectionName: CollectionNameString, slug: string): Promise<boolean> => {
+Utils.slugIsUsed = async (collectionName: CollectionNameString, slug: string, idToAllowOldSlugsFor?: string): Promise<boolean> => {
   const collection = getCollection(collectionName)
-  const existingUserWithSlug = await collection.findOne({$or: [{slug: slug},{oldSlugs: slug}]})
+  const existingUserWithSlug = await collection.findOne({$or: [
+    {slug: slug},
+    {$and: [{ oldSlugs: slug }, { _id: { $ne: idToAllowOldSlugsFor }}]}
```

Comment:
I have checked that this behaves the same as {$or: [{$slug: slug}, {$oldSlugs: slug}]} if idToAllowOldSlugsFor is undefined